src/libcore/slice/sort.rs

```diff
@@ -288,8 +288,8 @@ fn partition_in_blocks<T, F>(v: &mut [T], pivot: &T, is_less: &mut F) -> usize
 
         if start_r == end_r {
             // Trace `block_r` elements from the right side.
-            start_r = offsets_r.as_mut_ptr();
-            end_r = offsets_r.as_mut_ptr();
+            start_r = unsafe { offsets_r.get_mut().as_mut_ptr() };
```

Comment:
This code also looks like this should be invalid? At least, `offsets_r` is not necessarily initialized, right?